### PR TITLE
feat: add experiment and eventing to recommendations painted door exp

### PR DESCRIPTION
**Pull request by syedsajjadkazmii**
VAN-1604

**Description:**
- Setup painted door optimizely experiment
- Add eventing to painted door experiment

### CODE REVIEW

File: src/containers/LearnerDashboardHeader/CollapsedHeader/CollapseMenuBody.jsx [hunk 1]
```diff
@@ -99,21 +117,20 @@ export const CollapseMenuBody = ({ isOpen, isRecommendationsModalOpen, setIsReco
             </Button>
           </>
         )}
-        <ModalView isOpen={isRecommendationsModalOpen} onClose={setIsRecommendationsModalOpen} />
+        {(!experimentLoading && isPaintedDoorNavbarBtnVariation) && (
```
Comment (mubbsharanwar) [inline src/containers/LearnerDashboardHeader/CollapsedHeader/CollapseMenuBody.jsx:120]: Do we need this condition here? The button is already behind this condition on line 60 and the popup will appear when experiment is enabled and isOpen will true.

File: src/containers/LearnerDashboardHeader/ExpandedHeader/index.jsx [hunk 2]
```diff
@@ -73,7 +89,13 @@ export const ExpandedHeader = () => {
       </div>
 
       <AuthenticatedUserDropdown />
-      <ModalView isOpen={isRecommendationsModalOpen} onClose={toggleRecommendationsModal} />
+      {!experimentLoading && isPaintedDoorNavbarBtnVariation && (
```
Comment (mubbsharanwar) [inline src/containers/LearnerDashboardHeader/ExpandedHeader/index.jsx:92]: Same for here, do we need this condition?

File: src/widgets/LookingForChallengeWidget/index.jsx [hunk 3]
```diff
@@ -17,6 +17,11 @@ export const arrowIcon = (<Icon className="mx-1" src={ArrowForward} />);
 export const LookingForChallengeWidget = () => {
   const { formatMessage } = useIntl();
   const { courseSearchUrl } = reduxHooks.usePlatformSettingsData();
+
```
Comment (mubbsharanwar) [inline src/widgets/LookingForChallengeWidget/index.jsx:20]: Just asking for my understanding, is this component rendered when there would be no recommendations?

File: src/RecsPaintedDoorExpContext.jsx
Comment (mubbsharanwar) [inline src/RecsPaintedDoorExpContext.jsx:1]: Can we move this context under recommendations code like `RecommendationsPanel`?

File: src/containers/LearnerDashboardHeader/CollapsedHeader/CollapseMenuBody.jsx (hunk 1 above)
Comment (syedsajjadkazmii) [reply src/containers/LearnerDashboardHeader/CollapsedHeader/CollapseMenuBody.jsx:120]: you are right.

File: src/widgets/LookingForChallengeWidget/index.jsx (hunk 3 above)
Comment (syedsajjadkazmii) [reply src/widgets/LookingForChallengeWidget/index.jsx:20]: yes

File: src/RecsPaintedDoorExpContext.jsx
Comment (syedsajjadkazmii) [reply src/RecsPaintedDoorExpContext.jsx:1]: This is not just under the recommendations panel, this is also for headers, its basically a painted door exp context

**Review by mubbsharanwar — APPROVED** [review]

**Pull request merged**